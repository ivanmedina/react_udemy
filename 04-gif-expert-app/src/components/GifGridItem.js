import React from 'react';

export const GrifGridItem=({url, title})=>{
    // console.log(img);
    return(
        <div className="card">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}