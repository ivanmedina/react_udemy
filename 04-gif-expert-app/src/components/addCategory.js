import React, { useState } from 'react'
import propTypes from "prop-types";

export function AddCategory({setCategories}) {
    const [inputValue, setInputValue]=useState('');

    const handleInputChange=(e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit=(e)=>{
        e.preventDefault();
        if(inputValue.trim().length>2){
            setCategories(categorias=>[ inputValue, ...categorias]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type = "text"
                value = { inputValue }
                onChange = { handleInputChange}
            />
        </form>
        
    )
}

AddCategory.propTypes={
    setCategories: propTypes.func.isRequired
};

export default AddCategory
