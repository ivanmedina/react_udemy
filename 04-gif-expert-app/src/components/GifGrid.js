// import React, { useState, useEffect } from "react";
import React from 'react';
import {useFetchGifs} from '../hooks/useFetchGifs';
import { GifGridItem } from "./GifGridItem";
// import {getGifs} from '../helpers/getGifs'

export const GifGrid = ({category})=>{

    const {data,loading}= useFetchGifs(category);
    
    // const [gifs, setGifs]=useState([]);


    return(
        <>
            <h3 className="animate__animated animate__fadeIn">{ category }</h3>

            {loading && <p className="animate__animated animate__flash">cargando ...</p>}

            <div className="card-grid">
                    {
                        data.map( img =>(
                            <GifGridItem 
                                key={img.id}    
                                { ...img}
                            />
                        ))
                    }
            </div>
        </>
    )

}


