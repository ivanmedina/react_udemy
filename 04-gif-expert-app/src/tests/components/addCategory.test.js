import { shallow } from "enzyme";
import { AddCategory} from "../../components/addCategory";

describe('Pruebas para componente addCategory',()=>{
    
    // esto ...
    // const setCategories=()=>{};
    // es lo mismo que esto
    const setCategories=jest.fn();
    

    let wraper=shallow(<AddCategory setCategories={setCategories}/>);
    
    beforeEach(()=>{
        jest.clearAllMocks();
        wraper=shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('El componente debe mostrarse correctamente',()=>{

        expect(wraper).toMatchSnapshot();

    })

    test('Debe cambiar la caja de texto',()=>{

        const input=wraper.find('input');
        const value="Hola mundo";
        
        input.simulate('change',{target:{value}})

        expect(wraper.find('p').text().trim()).toBe(value);

    })

    test('Debe funcionar preventDefault en submit',()=>{
        wraper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe llamar al setCategories',()=>{
        
        const value="Hola mundo";

        // 1. simular el inputChange
        wraper.find('input').simulate('change',{target:{value}});

        // 2. simular el submit
        wraper.find('form').simulate('submit',{preventDefault(){}});
        
        // 3. setCategories se debe de haber llamado
        expect(setCategories).toHaveBeenCalled();

    })
})