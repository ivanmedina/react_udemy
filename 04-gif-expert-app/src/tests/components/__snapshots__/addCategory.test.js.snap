// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para componente addCategory El componente debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
